src/services/NotificationTypeService.js

```diff
@@ -0,0 +1,117 @@
+import { isMongoId } from 'validator';
+import { isNil, pick } from 'lodash';
+import { getListFilters } from '../utils/filter';
+import Type from '../models/type';
+import logger from '../utils/logger';
+import { ApiError } from '../utils/errors';
+
+const typesFilters = ['_id', 'version'];
+
+class NotificationTypeService {
+  getTypes = async req => {
+    try {
+      const {
+        query: { limit, ordering = '-createdAt' },
+        skip,
+      } = req;
+
+      let query = {};
+
+      const filters = getListFilters(typesFilters, req);
+      if (filters.length) {
+        query = { ...query, ...Object.assign(...filters) };
+      }
+
+      return Promise.all([
+        Type.find(query)
+          .limit(limit)
+          .skip(skip)
+          .sort(ordering)
+          .exec(),
+        Type.count(query),
+      ]);
+    } catch (err) {
+      logger.error(`[${this.constructor.name}.getTypes] Error: ${err}`);
+      throw err;
+    }
+  };
+
+  getType = async req => {
+    try {
+      const {
+        params: { id },
+      } = req;
+      const query = isMongoId(id) ? { _id: id } : { externalId: id };
```

Comment:
This is not appropriate for notification types because the id values are always MongoDB ids and they don't even have an externalId field. This code was in the user controller because user ids can be MongoDB ids or Cognito ids.

Reply:
done